caikit_nlp/modules/text_generation/peft_tgis_remote.py

```diff
@@ -63,6 +64,12 @@ def __init__(
             # loaded when using the local TGIS backend
             self._client = tgis_backend.get_client(base_model_name)
 
+            # Tell the backend to load all of the available prompt files
+            if prompt_artifacts:
+                tgis_backend.load_prompt_artifacts(
+                    base_model_name, prompt_cache_id, *prompt_artifacts
```

Comment:
am wondering if we need to specifically get the `prompt_artifacts` to the `load_prompt_artifacts` function, since the `load` function is going to be just copying over the file,  so instead of copying specific list of files, i.e `encoder.pt` and `decoder.pt` it might be better to copy the whole folder which will include the `config.yml` ? That way the shape of the "model" if looked inside TGIS container look similar to shape available on some storage system.

On the other side, if there are any artifacts, like "base_model" also stored in the "model_folder" passed on here, then the "load" can incur a lot of cost 🤔 

I guess, either way, changing this won't be API breaking, so may be we don't need to make decision on that, but may be for future 💭

Reply:
Yeah, I thought about this, but figured that this is "prompt specific" and in the future if TGIS exposes CRUD on prompts, we'll only be sending the prompt data and not the whole model (TGIS should not know or care about `caikit` formatted models). The point about models that contain the base model artifacts is valid too. If we did want to do this, I'd say we would want to nest `encoder.pt`/`decoder.pt` in a subdirectory like `prompt_artifacts`, but then the flat model directory would not be able to function as-is in TGIS since it would have an extra layer of name nesting. Given all this, I'm inclined to leave it as is.